library/node.go

```diff
@@ -51,10 +63,66 @@ func randomHex(n int) (string, error) {
 	return hex.EncodeToString(bytes), nil
 }
 
+func Init() *WakuInstance {
+	id := uint(len(wakuInstances))
```

Comment:
Need synchronization for accessing and updating this map, otherwise it can lead to issues. 

What if Init gets called in 2 threads and both return same len of the map?